src/request_collector/server.py

```diff
@@ -49,10 +49,14 @@ def stop(self) -> None:
         self.matrix_listener.join()
 
     def handle_message(self, message: Message) -> None:
-        if isinstance(message, RequestMonitoring):
-            self.on_monitor_request(message)
-        else:
-            log.debug("Ignoring message", message=message)
+        try:
+            if isinstance(message, RequestMonitoring):
+                self.on_monitor_request(message)
+            else:
+                log.debug("Ignoring message", message=message)
+        # add more advanced exception catching
+        except AssertionError as ex:
```

Comment:
This is one of the rare cases where I would go as far as doing `except Exception`. I don't see a way to make sure that maliciously constructed messages can't raise some exception we didn't think of and catching all those is a simple way to keep the RC running to accept valid messages.

However, the `reward_proof_signature is None` problems won't cause an exception here, but in https://github.com/raiden-network/raiden-services/blob/master/src/raiden_libs/matrix.py#L71 where it is properly handled. Maybe I'm being a bit too defensive, here. What do you thing @palango?

Reply:
I agree, let's catch everything that isn't handled before as a last resort kind of thing.

For the `reward_proof_signature is None`: I'd like to have a test at least to be sure in the future that it is handled.